moped-editor/src/views/projects/projectView/ProjectComments.js

```diff
@@ -154,10 +162,93 @@ const ProjectComments = () => {
     });
   };
 
+  /**
+   * Updates the type based on conditions
+   * @param {Number} typeId
+   */
+  const filterNoteType = typeId => setNoteType(Number(typeId));
+
+  /**
+   * Whenever noteType changes, we change the query conditions
+   */
+  useEffect(() => {
+    if (noteType === 0) {
+      setNoteTypeConditions({});
+    } else {
+      setNoteTypeConditions({
+        project_note_type: { _eq: noteType },
+      });
+    }
+    refetch();
+  }, [noteType, setNoteTypeConditions, refetch]);
+
+  // If the query is loading or data object is undefined,
+  // stop here and just render the spinner.
+  if (loading || !data) return <CircularProgress />;
+  if (error) return console.log(error);
+
   return (
     <Page title="Project Notes">
       <Container>
         <Grid container spacing={2}>
+          {/*New Note Form*/}
+          {!editingComment && (
+            <Grid item xs={12}>
+              <Card>
+                <CommentInputQuill
+                  noteText={noteText}
+                  setNoteText={setNoteText}
+                  editingComment={editingComment}
+                  commentAddLoading={commentAddLoading}
+                  commentAddSuccess={commentAddSuccess}
+                  submitNewComment={submitNewComment}
+                  submitEditComment={submitEditComment}
+                  cancelCommentEdit={cancelCommentEdit}
+                />
+              </Card>
+            </Grid>
+          )}
+          {/*First the Filter Buttons*/}
+          <Grid item xs={12}>
+            <FormControlLabel
+                className={classes.showButtonItem}
+                label="Show"
+                control={<span />}
+            />
+            <Button
+                className={classes.showButtonItem}
+                variant={noteType === 0 ? "contained" : "outlined"}
+                color="primary"
+                onClick={() => filterNoteType(0)}
+            >
+              All
+            </Button>
+            <Button
+                className={classes.showButtonItem}
+                variant={noteType === 1 ? "contained" : "outlined"}
+                color="primary"
+                onClick={() => filterNoteType(1)}
+            >
+              Internal Notes
+            </Button>
+            <Button
+                className={classes.showButtonItem}
+                variant={noteType === 2 ? "contained" : "outlined"}
+                color="primary"
+                onClick={() => filterNoteType(2)}
+            >
+              Status Updates
+            </Button>
+            <Button
+                className={classes.showButtonItem}
+                variant={noteType === 3 ? "contained" : "outlined"}
```

Comment:
It might be nice to pull the integer + id for note types from the DB directly via a lookup table. And then autocreate the buttons needed. That would give us flexibility if more note/comment types are adding in the future.

Reply:
Agreed it was tedious to write and to read, I'll refactor the code to have a re-usable button component instead.